Sources/SwiftParser/Specifiers.swift

```diff
@@ -619,24 +654,28 @@ extension Parser {
       throwsKeyword = throwsKw
     }
 
-    var unexpectedAfterThrowsLoopProgress = LoopProgressCondition()
-    while self.hasProgressed(&unexpectedAfterThrowsLoopProgress) {
+    if throwsKeyword != nil && self.at(.leftParen) && experimentalFeatures.contains(.typedThrows) {
+      thrownType = parseThrownType()
+    }
```

Comment:
IIUC this would also try to parse a thrown type for `try async(MyError)`, which I think is confusing. IMO we should parse the thrown type whenever we set the `throwsKeyword`